presto-raptorx/src/main/java/com/facebook/presto/raptorx/storage/LocalCleaner.java

```diff
@@ -0,0 +1,245 @@
+            waitJitterTime();
+            runLocalCleanupImmediately(local);
+        });
+
+        scheduler.scheduleWithFixedDelay(() -> {
+            waitJitterTime();
+            runLocalCleanup();
+        }, 0, localCleanerInterval.toMillis(), MILLISECONDS);
+    }
+
+    private void waitJitterTime()
+    {
+        try {
+            // jitter to avoid overloading database
+            long interval = this.localCleanerInterval.roundTo(SECONDS);
+            SECONDS.sleep(ThreadLocalRandom.current().nextLong(1, interval));
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    @VisibleForTesting
+    synchronized Set<Long> getLocalChunks()
+    {
+        return storageService.getStorageChunks();
+    }
+
+    private synchronized void runLocalCleanupImmediately(Set<Long> local)
+    {
+        try {
+            cleanLocalChunksImmediately(local);
+        }
+        catch (Throwable t) {
+            log.error(t, "Error cleaning local chunks");
+            localJobErrors.update(1);
+        }
+    }
+
+    private synchronized void runLocalCleanup()
+    {
+        try {
+            cleanLocalChunks();
+        }
+        catch (Throwable t) {
+            log.error(t, "Error cleaning local chunks");
+            localJobErrors.update(1);
+        }
+    }
+
+    @VisibleForTesting
+    synchronized void cleanLocalChunksImmediately(Set<Long> local)
+    {
+        // get chunks assigned to the local node
+        Set<Long> assigned = chunkManager.getNodeChunks(currentNode).stream()
+                .map(ChunkFile::getChunkId)
+                .collect(toSet());
+
+        // only delete local files that are not assigned
+        Set<Long> deletions = Sets.difference(local, assigned);
+
+        for (Long chunkId : deletions) {
+            deleteFile(storageService.getStorageFile(chunkId));
+        }
+
+        localChunksCleaned.update(deletions.size());
+        log.info("Cleaned %s local chunks immediately", deletions.size());
+    }
+
+    @VisibleForTesting
+    synchronized void cleanLocalChunks()
+    {
+        // find all files on the local node
+        Set<Long> local = getLocalChunks();
+
+        // get chunks assigned to the local node
+        Set<Long> assigned = chunkManager.getNodeChunks(currentNode).stream()
+                .map(ChunkFile::getChunkId)
+                .collect(toSet());
+
+        // un-mark previously marked files that are now assigned
+        for (Long chunkId : assigned) {
+            chunksToClean.remove(chunkId);
+        }
+
+        // mark all files that are not assigned
+        for (Long chunkId : local) {
+            if (!assigned.contains(chunkId)) {
+                chunksToClean.putIfAbsent(chunkId, ticker.read());
+            }
+        }
+
+        // delete files marked earlier than the clean interval
+        long threshold = ticker.read() - localCleanTime.roundTo(NANOSECONDS);
```

Comment:
I don't think `localCleanTime` is needed for Raptor v2. In Raptor, after we mark data as deleted, we wait for a period of time before we actually deleted the data from storage. We do it in this way because we don't know if there is any unfinished queries are referring the deleted data. So we wait for a long enough time (e.g. 4 hour) and assume the deleted data has 0 reference and is safe to be physically deleted. 

In Raptor V2, we actually have the ability to check if the deleted data is being referenced. In `TransactionManager` we have the `oldestActiveCommitId()` method which can help us define the right boundary for physical deletions.

Reply:
Yeah, let's put this in

Reply:
This is not a trivial change. Please check our quip design doc for details: https://fb.quip.com/jCZuA3Ubzrww#DfIACALLZZy

Reply:
@zhaotao1986 how can i access the design doc?

Reply:
A second thought about the `localCleanTime`.. Worker cannot get the oldest active commit id as the information is only available for the coordinator. So we have to use the previous approach, waiting for a long enough time before freeing the storage. 

We should have assertion in `TransactionManager` and throw exception when it's initialized on worker node.

Reply:
We keep V1's logic. Improve later